enhancements/olm-curated-catalog.md

```diff
@@ -0,0 +1,302 @@
+
+```
+
+This will mirror the entire catalog to the disconnected, local registry.  As catalogs begin to include hundreds of operators, versions of operators, versions of operands and multiple architectures, it will not be feasible to mirror the entire online catalog.
+
+### Curated Catalogs for Users
+Companies would like to be able to limit the list of operators to those that are endorsed or supported by the customer and their lines of business.
+
+Cluster administrators can build multiple catalog images and install them in different namespaces based on company policies.   Although, by itself, doesn't disable the ability to install other operators in other catalogs, it does provide a better user experience.
+
+
+### Goals
+
+* Define `opm` commands which can build or update a catalog index image from one or more other catalog index images using the following filters:
+  * Catalog Index Image
+  * Operator Packages
+  * Channels
+* Support `opm registry` and `opm image` use cases
+
+### Non-Goals
+
+
+
+## Proposal
+
+The basic approach will be to:
+
+- Generate catalog index container images that contain a subset of one or more existing catalog container images
+- Provide the ability to merge catalogs together, to allow periodic mirroring of catalogs that are updated over time.
+
+### Generate a curated index catalog
+
+`opm index filter` will be created with flags:
+- `-f, --filter-spec` - a path to a filter YAML file
+
+The following flags are inherited from the `opm index add` command:
+```
+  -i, --binary-image opm        container image for on-image opm command
+  -c, --container-tool string   tool to interact with container images (save, build, etc.). One of: [docker, podman] (default "podman")
+      --generate                if enabled, just creates the dockerfile and saves it to local disk
+  -h, --help                    help for add
+  -d, --out-dockerfile string   if generating the dockerfile, this flag is used to (optionally) specify a dockerfile name
+      --permissive              allow registry load errors
+  -t, --tag string              custom tag for container image being built
+```
+
+Examples:
+```sh
+$ opm index filter -t dockerreg.local/catalogs/company-catalog:latest -f ./filter.yaml
+```
+
+or
+```sh
+$ cat filter.yaml | opm index filter -t dockerreg.local/catalogs/company-catalog:latest  -f -
+```
+
+The `filterSpec` yaml specification is as follows:
+
+* `filterSpec` (array of objects): array of filter specification objects as follows
+  * `indexImage` (string): the catalog index image refererence
+  * `filter` (array of objects):  array of filter objects as follows:
+    * `package` (string) : the name of the operator package to include.
+    * `includedChannels` (array of string): the array of channel names to include from the package (Optional, defaults to all channels)
+    * `excludedChannels` (array of string): the array of channels to exclude (Optional, defaults to no excluded channels)
```

Comment:
I'm all for simplification.  I was expecting this response, and mostly wanted to include it for discussion and verifying that it will work in the future if needed.  We could move this into a future section.

The main reason for using it is to support use cases where updating the build policy may be difficult to do, or discovering channels may not be easy, but I don't really have any immediate requirements for `excludedChannnels`.

What do you think if we provided a regular expression instead of, or in addition to a matching list?

Reply:
I've been thinking about how to build separate catalogs to make sure "alpha" and "beta" aren't available channels when I promote a catalog into a production environment. I really appreciate the idea of at least "excludedChannels" especially for operators I'm developing within my own organization.